src/lib/ledger_catchup/catchup_worker.ml

```diff
@@ -0,0 +1,57 @@
+open Core
+open Async
+open Coda_base
+
+let dispatch rpc query peer =
+  Tcp.with_connection (Tcp.Where_to_connect.of_host_and_port peer)
+    ~timeout:(Time.Span.of_sec 1.) (fun _ r w ->
+      let open Deferred.Let_syntax in
+      match%bind Rpc.Connection.create r w ~connection_state:(fun _ -> ()) with
+      | Error exn -> return (Or_error.of_exn exn)
+      | Ok conn -> Rpc.Rpc.dispatch rpc conn query )
+
+module Make (Inputs : Inputs.S) = struct
+  open Inputs
+
+  let get_external_transitions ~frontier hash =
+    let open Option.Let_syntax in
+    let%map breadcrumb = Transition_frontier.find frontier hash in
+    Transition_frontier.path_map
+      ~f:Transition_frontier.Breadcrumb.transition_with_hash frontier
+      breadcrumb
+
+  module Worker_rpc = struct
+    type query = State_hash.t [@@deriving bin_io]
+
+    type response =
+      (Inputs.External_transition.t, State_hash.t) With_hash.t list option
```

Comment:
We should send the external transition without the hash. Someone could send the incorrect hash for a transition and screw us up, so we want to have the receiving client always calculate the hash themselves.